packages/fpdart/lib/src/exit.dart

```diff
@@ -0,0 +1,53 @@
+import 'effect.dart';
+
+typedef Exit<L, R> = Either<Cause<L>, R>;
+
+sealed class Cause<L> implements Error {
```

Comment:
Can you give me an idea of of interruption would work in dart? Is interruption something that the user triggers only manually (`.interrupt`) or is it possible for `Effect` to interrupt by itself in some cases?

Reply:
You would need to add an option to the async run methods that allow passing in some kind of abort signal (could be a dart Completer). Create a lightweight "handle" to the running Effect would also help, which you could call .interrupt() on.

Reply:
This would be about adding a `Completer` (or `Deferred`) optional parameter to `run*()`.

Then when executing `Effect` the function will first check the value inside `Deferred` and, if already present, return an `Interrupted` and stop execution.

Is my interpretation correct? What are some other usecases that I am missing?

Reply:
Also it would send a signal to the currently running Effect, so you could hook it up to something like dio's CancelToken.

Reply:
Added interruption using `Deferred` and `Context` (I see now how this all plays together 👌)